firepit/sqlstorage.py

```diff
@@ -960,7 +960,63 @@ def timestamped(
         else:
             res = qry
         return res
+    def getAllNestedObjectsWithAnAttributeOfSCO(
+            self,
+            viewname,
+            path=None,
+            value=None,
+            nameOfColumnTobeAppended='id',
+            limit=None,
+            run=True):
+        
+
+        # Something like this:
+        # select sco."{column}" as "{column}", obs."{ts}" as "{ts}"
+        #   from "{viewname}" sco
+        #     join __contains c on sco.id = c.target_ref
+        #     join "observed-data" obs on c.source_ref = obs.id
+        #   where sco."{column}" = {value};
 
+        qry = Query([
+            Table(viewname),
+            Join('__contains', 'id', '=', 'target_ref'),
+            Join('observed-data', 'source_ref', '=', 'id')
+        ])
+        table = viewname
+        if path:
+            if isinstance(path, (list, tuple)):
+                paths = path
+                column = None
+            else:
+                paths = [path]
+                column = path
+        else:
+            paths = []
+            column = None
+        proj = []
+        for p in paths:
+            if p == '*':
+                continue
+            joins, table, column = self.path_joins(viewname, None, p)
+            qry.extend(joins)
+            proj.append(Column(column, table, p))
+        if column and value is not None:
+            qry.append(Filter([Predicate(column, '=', value)]))
+        ts_col = Column(nameOfColumnTobeAppended, 'observed-data')
```

Comment:
You can't add another `id` column; there's already the base SCO type's `id` column.  You would have to alias it to something else, like `x_contained_by` or `x_observation_id` etc.